CSharpBashInterpreter.Tests/CommandTests/GrepTests.cs

```diff
@@ -0,0 +1,162 @@
+        try
+        {
+            var testParagraphs = Lorem.Paragraphs(5).ToList();
+            var testText = String.Join(Environment.NewLine, testParagraphs);
+            File.WriteAllText(tempFileName, testText);
+
+            var grepCommandExecutable = new GrepCommandRepresentation().Build(new[] { "grep", testParagraphs[2].Split().First(), tempFileName });
+            var pipe = new Pipe();
+            using var reader = new StreamReader(pipe.Reader.AsStream());
+            var streams = new StreamSet
+            {
+                OutputStream = pipe.Writer.AsStream(),
+            };
+            grepCommandExecutable.ExecuteAsync(streams).Result.Should().Be(0);
+            reader.ReadToEndAsync().Result.Should().Contain("3:"+testParagraphs[2]);
+        }
+        finally
+        {
+            File.Delete(tempFileName);
+        }
+    }
+
+
+    [Fact]
+    public void GrepTestIFlag()
+    {
+        var tempFileName = Path.GetTempFileName();
+        try
+        {
+            var testWord = Lorem.GetFirstWord();
+            var testText = String.Join(Environment.NewLine,
+                new[]
+                {
+                    testWord,
+                    testWord.ToUpper(),
+                    testWord.ToLower(),
+                    char.ToUpper(testWord[0]) + new string(testWord.Skip(1).ToArray())
+                });
+            File.WriteAllText(tempFileName, testText);
+
+            var grepCommandExecutable = new GrepCommandRepresentation().Build(new[] { "grep", "-i", testWord.ToUpper(), tempFileName });
+            var pipe = new Pipe();
+            using var reader = new StreamReader(pipe.Reader.AsStream());
+            var streams = new StreamSet
+            {
+                OutputStream = pipe.Writer.AsStream(),
+            };
+            grepCommandExecutable.ExecuteAsync(streams).Result.Should().Be(0);
+            reader.ReadToEndAsync().Result.Should().ContainAll(testWord, testWord.ToLower(), testWord.ToUpper());
+        }
+        finally
+        {
+            File.Delete(tempFileName);
+        }
+    }
+
+
+    [Fact]
+    public void GrepTestWFlag()
+    {
+        var tempFileName = Path.GetTempFileName();
+        try
+        {
+            var testWord = Lorem.GetFirstWord();
+            var testText = string.Join(Environment.NewLine, testWord, testWord+testWord+testWord, testWord+testWord);
+            File.WriteAllText(tempFileName, testText);
+
+            var grepCommandExecutable = new GrepCommandRepresentation().Build(new[] { "grep", "-w", testWord, tempFileName });
+            var pipe = new Pipe();
+            using var reader = new StreamReader(pipe.Reader.AsStream());
+            var streams = new StreamSet
+            {
+                OutputStream = pipe.Writer.AsStream(),
+            };
+            grepCommandExecutable.ExecuteAsync(streams).Result.Should().Be(0);
+            var result =reader.ReadToEndAsync().Result;
```

Comment:
```suggestion
            var result = reader.ReadToEndAsync().Result;
```
Что-то у вас беда с пробелами

Reply:
Fixed